scripts/cluster-upgrade.sh

```diff
@@ -310,18 +310,24 @@ upgrade() {
     log $OCM_NAME "upgrade" "ClusterVersion on $TO"
     log $OCM_NAME "upgrade" "checking node versions"
 
-    # fun fact!  after clusterversion says it is done, individual nodes could still be updated.
-    MACHINE_COUNT_ALL=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get nodes --no-headers | wc -l)
+    MASTER_MACHINEPOOL_TOTAL_COUNT=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get machineconfigpools -o json  | jq -r ".items[] | select(.metadata.name == \"master\" ) | .status.machineCount")
+    MASTER_MACHINEPOOL_UPDATED_COUNT=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get machineconfigpools -o json  | jq -r ".items[] | select(.metadata.name == \"master\" ) | .status.updatedMachineCount")
 
-    MACHINE_COUNT_UPGRADED=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get nodes -o json | jq -r ".items[].status | select(.annotations[\"machineconfiguration.openshift.io/currentConfig\"] == .annotations[\"machineconfiguration.openshift.io/desiredConfig\"]) | .conditions[] | select(.type == \"Ready\" and .status == \"True\") | .reason" | wc -l)
+    while [ $MASTER_MACHINEPOOL_UPDATED_COUNT -lt $MASTER_MACHINEPOOL_TOTAL_COUNT ];
+    do
+        # wait for all masters to be upgraded
+        log $OCM_NAME "upgrade" "master nodes upgraded: $MASTER_MACHINEPOOL_UPDATED_COUNT/$MASTER_MACHINEPOOL_TOTAL_COUNT"
+        sleep 60
+    done
```

Comment:
If the initial check determines that not everything is upgraded, this will spin forever, won't it? We need to recalculate (repeat L313-4 -- though preferably pull that out into a helper function so it's not copy/pasted) inside the loop.

Reply:
From a point of view, running a check loop over the master machineconfigpool won't be necessary as the CVO would already handle that for the control plane. I checked this with MCO team yesterday. So, we can even simply introduce the check for worker pool check the "Updated" = True field to happen. For the timeout variable, will need to factor in the time needed for rolling update to apply change, mark node unschedulable and reboot the node. I think that would be 5-10 minutes per node. So can set it for cluster accordingly on a safer note.

Reply:
I think it's safe to include this check, I will put  in a PR to update the upgraded count soon, after the sleep.